app/lib/router/router.dart

```diff
@@ -777,22 +779,11 @@ List<RouteBase> makeRoutes(Ref ref) {
                     .switchTo(const Key('tasks'));
                 return NoTransitionPage(
                   key: state.pageKey,
-                  child: SpaceTasksPage(
-                    spaceIdOrAlias: state.pathParameters['spaceId']!,
-                  ),
-                );
-              },
-            ),
-            GoRoute(
-              name: Routes.space.name,
-              path: Routes.space.route,
-              redirect: authGuardRedirect,
-              pageBuilder: (context, state) {
-                tabKeyNotifier.switchTo(const Key('overview'));
-                return NoTransitionPage(
-                  key: state.pageKey,
-                  child: SpaceOverview(
+                  child: SpaceShell(
```

Comment:
I am not really digesting it of switching to normal nested navigation view with a common wrapper over all the routes, its solving one thing but raises other many. We are effectively making it re-render the whole page instead of  nested one and because tab view is build on demand, there's no persistence of state between tabs with this change. ```ShellRoute``` keeps that state and I would like to keep as it is there but maybe refactoring it solely would be something I would vouch for.

Reply:
The problem with keeping the old style is that it was impossible to get the proper navigation flow to work. And it makes sense once you read up on how ShellRoutes are _meant_ to be used: they are understood as two separate routes (even on the same screen) - e.g. have a split screen and each its own router with independent routing. The way we use it isn't really how that is meant to be used ... it isn't meant for hierarchy nested routes ... that's what the `GoRoute(routes: [...])` is meant to do...

Reply:
Alright, I think we can merge as it is and can work on in a separate PR.